orchestrator/careplancontributor/service.go

```diff
@@ -111,57 +111,42 @@ func (s Service) confirm(localFHIR fhirclient.Client, serviceRequestRef string,
 		return nil, fmt.Errorf("failed to create Task: %w", err)
 	}
 
-	// Create channels for polling the task, 1 channel for the result and 1 for an error
-	resultChan := make(chan *fhir.Task)
-	errorChan := make(chan error)
-
 	// Start polling in a new goroutine so that the code continues to the select below
-	go s.pollTaskStatus(*task.Id, resultChan, errorChan)
-
-	// Block until either the task is accepted or an error occurs (error after 10 attempts)
-	select {
-	case result := <-resultChan:
-		log.Info().Msgf("Task/%s accepted: %+v\n", *result.Id, result)
-		enrichedTask, err := s.handleAcceptedTask(task, serviceRequest, &patient)
-		return enrichedTask, err
-	case err := <-errorChan:
-		fmt.Printf("error while polling task %s: %v\n", *task.Id, err)
-		return nil, err
+	err = s.pollTaskStatus(*task.Id)
+
+	if err != nil {
+		return nil, fmt.Errorf("error while polling task %s: %v\n", *task.Id, err)
 	}
+
+	return s.handleAcceptedTask(task, serviceRequest, &patient)
 }
 
 // pollTaskStatus polls the task status until it is accepted, an error occurs or reaches a max poll amount.
-func (s Service) pollTaskStatus(taskID string, resultChan chan<- *fhir.Task, errorChan chan<- error) {
-	ticker := time.NewTicker(2 * time.Second) // Poll every 2 seconds
-	defer ticker.Stop()
-
-	maxAttempts := 10
-	attempts := 0
-
-	for range ticker.C {
-
-		if attempts >= maxAttempts {
-			err := fmt.Errorf("maximum number of attempts reached for Task/%s", taskID)
-			log.Error().Msgf("polling Task/%s failed - error: [%s]", taskID, err)
-			errorChan <- err
-			return
+func (s Service) pollTaskStatus(taskID string) error {
+	pollInterval := 2 * time.Second
+	maxPollingDuration := 20 * time.Second
+	startTime := time.Now()
+
+	for {
+		if time.Since(startTime) >= maxPollingDuration {
+			return fmt.Errorf("maximum polling duration of %d seconds reached for Task/%s", maxPollingDuration, taskID)
```

Comment:
I think it will print it in nanoseconds now, better:
```suggestion
			return fmt.Errorf("maximum polling duration of %s reached for Task/%s", maxPollingDuration, taskID)
```

(will print `20s`)